src/unix/haiku/native.rs

```diff
@@ -446,6 +520,50 @@ cfg_if! {
                 }
             }
         }
+
+        #[cfg(libc_union)]
+        impl PartialEq for __c_anonymous_cpu_topology_info_data {
+            fn eq(&self, other: &__c_anonymous_cpu_topology_info_data) -> bool {
+                unsafe {
+                self.root == other.root
+                    || self.package == other.package
+                    || self.core == other.core
+                }
+            }
```

Comment:
This is unsound, as the fields may be uninitialized. Safe functions must not access union fields that may be uninitialized.
Cc https://github.com/rust-lang/libc/issues/2816